include/deal.II/meshworker/scratch_data.h

```diff
@@ -0,0 +1,1301 @@
+      const unsigned int subface_no);
+
+    /**
+     * Get the currently initialized FEValues.
+     *
+     * This function will return the internal FEValues if the
+     * reinit(cell) function was called last. If the reinit(cell, face_no)
+     * function was called, then this function returns the internal
+     * FEFaceValues, and if the reinit(cell, face_no, subface_no) function was
+     * called (with a valid @p subface_no argument), it returns the internal
+     * FESubfaceValues object.
+     */
+    const FEValuesBase<dim, spacedim> &
+    get_current_fe_values() const;
+
+    /**
+     * Get the currently initialized neighbor FEValues.
+     *
+     * This function will return the neighbor FEValues if the
+     * reinit_neighbor(cell) function was called last. If the
+     * reinit_neighbor(cell, face_no) function was called, then this function
+     * returns the internal neighbor FEFaceValues, and if the
+     * reinit_neighbor(cell, face_no, subface_no) function was
+     * called (with a valid @p subface_no argument), it returns the internal neighbor
+     * FESubfaceValues object.
+     */
+    const FEValuesBase<dim, spacedim> &
+    get_current_neighbor_fe_values() const;
+
+    /**
+     * Store internally a copy of the given object. The copied object is
+     * owned by this class, and is accessible via reference through the
+     * get_object_with_name() method.
+     */
+    template <typename Type>
+    void
+    add_copy(const Type &entry, const std::string &name);
+
+    /**
+     * Add a reference to an already existing object. The object is not
+     * owned by this class, and the user has to guarantee that the
+     * referenced object lives longer than this class instance. The stored
+     * reference is accessible through the get_object_with_name() method.
+     */
+    template <typename Type>
+    void
+    add_reference(Type &entry, const std::string &name);
+
+    /**
+     * Return a reference to the object with given name. This function
+     * throws an exception if either an object with the given name is not
+     * stored in this class, or if the object with the given name is not of
+     * compatible type.
+     */
+    template <typename Type>
+    Type &
+    get_object_with_name(const std::string &name);
+
+    /**
+     * Return a constant reference to the object with the given name. This
+     * function throws an exception if either the name does not exist or if the
+     * object stored in the class with the given name cannot be converted to
+     * the given type.
+     */
+    template <typename Type>
+    const Type &
+    get_object_with_name(const std::string &name) const;
+
+    /**
+     * Find out if we store an object with given name.
+     */
+    bool
+    stores_object_with_name(const std::string &name) const;
+
+    /**
+     * Return the quadrature points of the internal FEValues object.
+     */
+    const std::vector<Point<spacedim>> &
+    get_quadrature_points() const;
```

Comment:
same here -- this class has a lot of member functions, it would be nice to group them for doxygen